### ISSUE TITLE
#3643 breaks binary wheels on CPU-only machine

### ISSUE THREAD

**Issue by hcho3** [dmlc/xgboost#3746]
After PR #3643 was merged to master, artifact builds are experiencing failure due to memory error: https://xgboost-ci.net/blue/organizations/jenkins/restricted-xgboost/detail/master/21/pipeline

Diagnosis: the generated Python binary wheels (`xgboost-*-py2-none-any.whl`) are no longer compatible with CPU-only containers. I'll try to re-produce this problem manually on my end (by building the wheel on a P2 instance and copying it over to an C5 instance).

@trivialfis @RAMitchell @canonizer 

Backtrace:
```
/lib64/libc.so.6(+0x75e5e)[0x7efd9af9de5e]
/lib64/libc.so.6(+0x78cf0)[0x7efd9afa0cf0]
/home/ubuntu/.local/lib/python2.7/site-packages/xgboost/./lib/libxgboost.so(_ZN7xgboost16HostDeviceVectorIfED1Ev+0x12e)[0x7efd78882dbe]
/home/ubuntu/.local/lib/python2.7/site-packages/xgboost/./lib/libxgboost.so(_ZN7xgboost4data15SimpleCSRSourceD1Ev+0x9a)[0x7efd78681bda]
/home/ubuntu/.local/lib/python2.7/site-packages/xgboost/./lib/libxgboost.so(XGDMatrixSliceDMatrix+0x8e5)[0x7efd7867ffa5]
/opt/python/lib/python2.7/lib-dynload/../../libffi.so.6(ffi_call_unix64+0x4c)[0x7efd95c43ec0]
```

**Comment by hcho3** [comment]
The CI tests did not catch this problem before merging because wheel building isn't being currently tested for pull requests since #3736. To prevent similar problems in the future, I will add back wheel building tests to pull requests.

**Comment by trivialfis** [comment]
I will try to fix it as soon as possible.

**Comment by hcho3** [comment]
Thanks. I'll also take a look if I get a chance.

**Comment by hcho3** [comment]
Fixed by #3751.
